frontend/components/instruments/StepSequencer.js

```diff
@@ -0,0 +1,229 @@
+import React, {useState, useEffect, useRef} from "react";
+import PropTypes from "prop-types";
+import SamplePlayer from "./SamplePlayer";
+import STYLES from "./StepSequencer.module.scss";
+
+const Step = ({sample, audioContext, colIsPlaying, vol}) => {
+    const [isActive, setIsActive] = useState(false);
+
+    const handleClick = () => {
+        setIsActive(!isActive);
+    };
+
+    // Pick button style depending on if sample is loaded
+    const btnStyle =
+        isActive
+            ? STYLES.activeStep
+            : STYLES.inactiveStep;
+
+    return (<>
+        <button
+            className={btnStyle}
+            onClick={handleClick}
+        />
+        <SamplePlayer
+            sample={sample}
+            shouldPlay={colIsPlaying && isActive}
+            loop={false}
+            volume={vol}
+            audioContext={audioContext}
+        />
+    </>);
+};
+Step.propTypes = {
+    sample: PropTypes.string,
+    colIsPlaying: PropTypes.bool,
+    audioContext: PropTypes.object,
+};
+
+const Metronome = ({sample, audioContext, colIsPlaying}) => {
+    // Pick button style depending on if sample is loaded
+    const btnStyle =
+        colIsPlaying
+            ? STYLES.isPlaying
+            : STYLES.notPlaying;
+
+    return (<>
+        <button
+            className={btnStyle}
+        />
+        <SamplePlayer
+            sample={sample}
+            shouldPlay={colIsPlaying}
+            loop={false}
+            volume={100}
+            audioContext={audioContext}
+        />
+    </>);
+};
+Metronome.propTypes = {
+    sample: PropTypes.string,
+    colIsPlaying: PropTypes.bool,
+    audioContext: PropTypes.object,
+};
+
+const Row = ({sample, audioContext, rowIndex, numSteps, playbackPosition, isPlaying}) => {
+    const [volume, setVolume] = useState(100);
+
+    const handleUpdateVolume = (event) => {
+        event.preventDefault();
+        if (!isPlaying){ // can only move slider when paused
+            setVolume(parseInt(event.target.value));
+        }
+    };
+
+    return (<>
```

Comment:
We don't need the `<>` and `</>` tags here: `<>` is a shortcut for `<React.Fragment>`, which is only required as a wrapper if your component returns more than one element. In this case, the outermost element is a single `<div>`, so you can just return that. See https://reactjs.org/docs/fragments.html